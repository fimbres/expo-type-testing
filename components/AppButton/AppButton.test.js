import React from "react";
import renderer from "react-test-renderer";

import { AppButton } from ".";

let tree;

describe("<AppButton />", () => {
  beforeEach(() => {
    tree = renderer
      .create(
        <AppButton
          title="Test Me"
          size="md"
          onPress={() => console.log("onPress")}
        />
      )
      .toJSON();
  });

  it("is defined", () => {
    expect(tree).toBeDefined();
  });

  it("has 1 child", () => {
    expect(tree.children.length).toBe(1);
  });

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
