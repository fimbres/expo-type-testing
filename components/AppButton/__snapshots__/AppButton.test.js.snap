// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppButton /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#334FFA",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "paddingVertical": 17,
      },
      false,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#020202",
          "fontFamily": "avenir-black",
          "fontSize": 12,
          "fontWeight": "800",
          "lineHeight": 16.39,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Test Me
  </Text>
</View>
`;
